### PR TITLE
8283287: ClassLoader.c cleanups

### PR DESCRIPTION
**Pull request by backwaterred**
As mentioned in the issue, I'd like to perform the following tidying on ClassLoader.c

- Alphabetize includes.
- Replace 'if (ptr == 0)' with 'if (ptr == NULL)'.
- Replace 'return 0' with 'return NULL'.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283287](https://bugs.openjdk.java.net/browse/JDK-8283287): ClassLoader.c cleanups

### Reviewers
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to 920e76fb471583a2a692bff2b0f44a7ce91d7dd7
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to 920e76fb471583a2a692bff2b0f44a7ce91d7dd7
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**) ⚠️ Review applies to 2fd5b3680738263071c6671c2732b8bdf53b15cd

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7846/head:pull/7846` \
`$ git checkout pull/7846`

Update a local copy of the PR: \
`$ git checkout pull/7846` \
`$ git pull https://git.openjdk.java.net/jdk pull/7846/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7846`

View PR using the GUI difftool: \
`$ git pr show -t 7846`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7846.diff">https://git.openjdk.java.net/jdk/pull/7846.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/native/libjava/ClassLoader.c [hunk 1]
```diff
@@ -23,16 +23,17 @@
  * questions.
  */
 
-#include <stdlib.h>
 #include <assert.h>
+#include <errno.h>
```
Comment (tstuefe) [inline src/java.base/share/native/libjava/ClassLoader.c:27]: Whats errno.h for?

**Review by tstuefe — COMMENTED** [review]
Looks good in general. Question inline.

File: src/java.base/share/native/libjava/ClassLoader.c (hunk 1 above)
Comment (backwaterred) [reply src/java.base/share/native/libjava/ClassLoader.c:27]: Thanks for catching that. It was for [the other PR](https://github.com/openjdk/jdk/pull/7829).

**Review by tstuefe — APPROVED** [review]
All good.

Cheers, Thomas

**Review by AlanBateman — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]

**Comment by backwaterred** [comment]
Please do not sponsor yet.

**Comment by backwaterred** [comment]
Made some additional readability changes that were requested after #7829 closed. They felt appropriate here since it's already a cleanup PR.

All changes complete. Please sponsor when ready.

**Comment by backwaterred** [comment]
Thanks :-)

/integrate

**Pull request closed without merging**